app/src/main/java/com/proptit/budgetbuddy/presentation/ui/more/category/add/AddNewCategoryFragment.kt

```diff
@@ -0,0 +1,104 @@
+import androidx.navigation.fragment.findNavController
+import androidx.recyclerview.widget.GridLayoutManager
+import com.bumptech.glide.Glide
+import com.proptit.budgetbuddy.databinding.FragmentAddNewCategoryBinding
+import com.proptit.budgetbuddy.domain.model.CategoryType
+import com.proptit.budgetbuddy.presentation.ui.more.category.CategoryIconAdapter
+import com.proptit.budgetbuddy.presentation.util.Constant
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.launch
+
+@AndroidEntryPoint
+class AddNewCategoryFragment : Fragment() {
+
+    private var _binding: FragmentAddNewCategoryBinding? = null
+    private val binding get() = _binding!!
+
+
+    private val addNewCategoryViewModel: AddNewCategoryViewModel by viewModels()
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        _binding = FragmentAddNewCategoryBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        initRecyclerViewCategoryIcons()
+        initBehavior()
+    }
+
+    private fun initBehavior() {
+        binding.editTextCategoryName.setOnClickListener {
+
+        }
+
+        binding.imageViewDone.setOnClickListener {
+            val categoryName = binding.editTextCategoryName.text.toString()
+            if (categoryName.isEmpty()) {
+                Toast.makeText(requireContext(), "Category name is empty", Toast.LENGTH_SHORT)
+                    .show()
+            } else {
+                val type = arguments?.getInt(Constant.CATEGORY_TYPE)!!
+                val categoryType = when (type) {
+                    0 -> CategoryType.EXPENSES
+                    else -> CategoryType.INCOME
+                }
+                addNewCategoryViewModel.addNewCategory(
+                    categoryName,
+                    addNewCategoryViewModel.iconUrl.value,
+                    categoryType
+                )
+//                findNavController().previousBackStackEntry?.savedStateHandle?.set("currentTabPosition", type)
+                findNavController().popBackStack()
+            }
+        }
+    }
+
+    private fun initRecyclerViewCategoryIcons() {
+        val categoryIconAdapter = CategoryIconAdapter(
+            onItemClick = { it ->
+                addNewCategoryViewModel.updateIconUrl(it)
+            }
+        )
+        binding.recyclerViewCategoryIcon.apply {
+            layoutManager = GridLayoutManager(requireContext(), 4)
+            adapter = categoryIconAdapter
+        }
+
+        lifecycleScope.launch {
+            viewLifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
+                categoryIconAdapter.setCategoryIcons(Constant.getCategoryIconLists())
+            }
+        }
+        lifecycleScope.launch {
```

Comment:
```suggestion
       addNewCategoryViewModel.iconUrl.onEach {
            Glide.with(requireContext()).load(Uri.parse(it))
                .into(binding.imageViewNewCategory)
        }.flowWithLifecycle(viewLifecycleOwner.lifecycle, Lifecycle.State.STARTED)
            .launchIn(viewLifecycleOwner.lifecycleScope)
```

I much prefer it this way.